framework/src/iotbus/iotbus_gpio.c

```diff
@@ -528,7 +594,8 @@ int iotbus_gpio_write(iotbus_gpio_context_h dev, int value)
 	}
 
 	handle = (struct _iotbus_gpio_s *)dev->handle;
-	ret = write(handle->fd, buf, snprintf(buf, sizeof(buf), "%d", !!value));
+	buf = !!value;
+	ret = write(handle->fd, &buf, 1);
```

Comment:
If write returns size of buffer, then it means that it worked properly. otherwise, it may an error.
Also IOTBUS_ERROR_UNKNOWN is not acceptable, in my opinion.